sugondat/shim/src/sugondat_rpc/mod.rs

```diff
@@ -342,6 +350,27 @@ pub struct Block {
     pub blobs: Vec<Blob>,
 }
 
+impl TryFrom<(Header, Vec<sugondat_subxt::ExtrinsicDetails>)> for Block {
+    type Error = anyhow::Error;
+    fn try_from(
+        value: (Header, Vec<sugondat_subxt::ExtrinsicDetails>),
+    ) -> Result<Self, Self::Error> {
+        let (header, extrinsics) = value;
+        let tree_root = tree_root(&header).ok_or_else(err::no_tree_root)?;
+        let timestamp = extract_timestamp(&extrinsics)?;
+        let blobs = extract_blobs(extrinsics);
+        tracing::debug!(?blobs, "found {} blobs in block", blobs.len());
+        Ok(Block {
+            number: header.number as u64,
+            hash: header.hash().0,
+            parent_hash: header.parent_hash.0,
+            tree_root,
+            timestamp,
+            blobs,
+        })
+    }
+}
```

Comment:
I am feeling rather strong about putting any significant logic in From/TryFrom. Those should be mechanical, pure functions. Moreover, IMO, `TryFrom` should have tight errors (i.e. listed in an `enum`). 

At the same time there is no benefit of using `try_from` in the first place.

I would suggest just using an associated function.